Add `require "rails"` to generated engines

Earlopain opened the pull request:
### Motivation / Background

Normally, gems get loaded in `application.rb` after `require "rails"` has already been run, so there is not much of an issue.

However, trying to require the gem in isolation will fail since the all sorts of things have not been set up yet (like the `Rails` constant).

### Additional information

I encountered this with `good_job` and `bundle console` but I don't think it's a very rare issue. `solid_queue` for example runs into the same thing (and also other problems).

If `rails` comes first in the gemfile, it works. If it comes after, it doesn't. Additionally, if you cherry-pick your frameworks and don't pull in all of rails, the gem order doesn't seem to matter at all.

cc @bensheldon 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

rafaelfranca reviewed (changes requested):
I like the idea, but we should not be using Rails code in plugin code outside of the Railtie. Can you move the require to the railtie/engine file?

If you have a Rails plugin and you need to interface with `Rails` and constants defined in the `railties` gem, it should be done in the Railtie of your gem.

Earlopain committed:
Add `require "rails"` to generated engines

Normally, gems get loaded in `application.rb` after `require "rails"` has already been run, so usually there is not much of an issue.

However, trying to require the gem in isolation will fail since the all sorts of things have not been set up yet (like the `Rails` constant).
This can be observed when running `bundle console` since it doesn't go through the rails initialization process.

Earlopain commented:
That sounds fine to me, I moved it over there. Thanks for the review.
